indico/modules/events/papers/models/revisions.py

```diff
@@ -152,3 +152,20 @@ def paper(self):
     @paper.setter
     def paper(self, paper):
         self._contribution = paper.contribution
+
+    def has_user_reviewed(self, user, review_type=None):
+        from indico.modules.events.papers.models.reviews import PaperReviewType
+        if review_type:
+            for review in self.reviews:
+                if review.user == user and review.type == PaperReviewType[review_type]:
+                    return True
+            return False
+        else:
+            user_reviews = 0
+            for review in self.reviews:
+                if review.user == user:
+                    user_reviews += 1
+            if user in self._contribution.paper_layout_reviewers and user in self._contribution.paper_content_reviewers:
+                return user_reviews == 2
```

Comment:
I'd go for something like this and then check whether these two vars aren't falsy:

```python
layout_review = next((x for x in self.reviews if review.user == user and review.type == whatever), None)
content_review = next((x for x in self.reviews if review.user == user and review.type == whatever), None)
```